.github/workflows/release.yml

```diff
@@ -0,0 +1,58 @@
+name: aiodynamo release workflow
+
+on:
+  release:
+    types: [released] # only triggered on release, not pre-release or draft release
+
+permissions:
+  contents: read
+  id-token: write
```

Comment:
That's unusual, what does it really do / why is it needed?

Reply:
It's required for PyPI trusted publishing
https://github.com/pypa/gh-action-pypi-publish#trusted-publishing